core/src/main/java/org/apache/accumulo/core/clientImpl/InstanceOperationsImpl.java

```diff
@@ -564,7 +564,7 @@ private Set<ServerId> getServers(ServerId.Type type, Predicate<String> resourceG
           String location = null;
           if (sld.isPresent()) {
             location = sld.orElseThrow().getAddressString(ThriftService.MANAGER);
-            if (addressSelector.getPredicate().test(location)) {
+            if (location != null && addressSelector.getPredicate().test(location)) {
```

Comment:
This probably could have been part of the predicate?

Reply:
It could have been, but then any predicate could allow a null to return true causing a NPE. If we didn't make this one change here, then we would need to add not null checks in all current and future address predicates.